pkg/server/serveroption/option.go

```diff
@@ -50,3 +89,35 @@ func WithMaxFlows(size int) Option {
 		return nil
 	}
 }
+
+// WithCiliumDaemon provides access to the Cilium daemon via downcast
+func WithCiliumDaemon(daemon interface{}) Option {
+	return func(o *Options) error {
+		o.CiliumDaemon = daemon
+		return nil
+	}
+}
+
+// WithAfterServerInit adds a new callback to be invoked after server initialization
+func WithAfterServerInit(cb AfterServerInit) Option {
+	return func(o *Options) error {
+		o.AfterServerInit = append(o.AfterServerInit, cb)
+		return nil
+	}
+}
+
+// WithBeforeDecoding adds a new callback to be invoked before decoding
+func WithBeforeDecoding(cb BeforeDecoding) Option {
+	return func(o *Options) error {
+		o.BeforeDecoding = append(o.BeforeDecoding, cb)
+		return nil
+	}
+}
+
+// WithAfterDecoding adds a new callback to be invoked after decoding
+func WithAfterDecoding(cb AfterDecoding) Option {
+	return func(o *Options) error {
+		o.AfterDecoding = append(o.AfterDecoding, cb)
+		return nil
+	}
+}
```

Comment:
You're missing the third part of this boilerplate which allows pure functions to act as options. Good example of that is `http.HandlerFunc`. It makes testing easier, and simple usages don't require a conrete type definition to conform to one of the interfaces.

Reply:
Sure, I can add this back!